ufo/ufo-profiler.c

```diff
@@ -136,7 +139,15 @@ ufo_profiler_call (UfoProfiler    *profiler,
         g_array_append_val (priv->event_array, row);
     }
     else {
-        cl_err = clEnqueueNDRangeKernel (command_queue, kernel, work_dim, NULL, global_work_size, local_work_size, 0, NULL, NULL);
+        cl_err = clEnqueueNDRangeKernel (command_queue, kernel, work_dim, NULL, global_work_size, local_work_size, 0, NULL, &event);
+    }
+
+    if (block) {
+        cl_err = clWaitForEvents (1, &event);
+        UFO_RESOURCES_CHECK_CLERR (cl_err);
+        if (!priv->trace) {
```

Comment:
I don't know what the tracing does with the event, so I release only if tracing is disabled. @matze please check.

Reply:
Copy of my previous answer: It keeps it alive for later analysis. You never used `ufo-launch -t ...` and then ran `ufo-prof opencl*.json`? But yes, if tracing is disabled you can safely release it.